tests/Replication2/Storage/LogPersistorTest.cpp

```diff
@@ -0,0 +1,376 @@
+#include "velocypack/Slice.h"
+
+#include <absl/crc/crc32c.h>
+#include <chrono>
+#include <cstdint>
+#include <cstring>
+#include <immer/flex_vector_transient.hpp>
+#include <ios>
+#include <memory>
+#include <variant>
+
+using namespace ::testing;
+using namespace arangodb::replication2;
+using namespace arangodb::replication2::replicated_log;
+
+namespace {
+auto makeNormalLogEntry(std::uint64_t term, std::uint64_t index,
+                        std::string payload) {
+  return InMemoryLogEntry{LogEntry{LogTerm{term}, LogIndex{index},
+                                   LogPayload::createFromString(payload)}};
+}
+
+auto makeMetaLogEntry(std::uint64_t term, std::uint64_t index,
+                      LogMetaPayload payload) {
+  return InMemoryLogEntry{LogEntry{LogTerm{term}, LogIndex{index}, payload}};
+}
+}  // namespace
+
+namespace arangodb::replication2::storage::wal::test {
+
+struct LogPersistorTest : ::testing::Test {
+  void SetUp() override {
+    auto file = std::make_unique<InMemoryFileWriter>(buffer);
+    EXPECT_CALL(*fileManager, createWriter(_))
+        .WillOnce(Return(std::move(file)));
+    persistor = std::make_unique<LogPersistor>(LogId{42}, fileManager);
+  }
+
+  void insertEntries() {
+    log = InMemoryLog{}.append({
+        makeNormalLogEntry(1, 1, "blubb"),
+        makeNormalLogEntry(1, 2, "dummyPayload"),
+        makeMetaLogEntry(
+            1, 3,
+            LogMetaPayload::withPing(
+                "message",
+                // the timepoint is serialized as seconds, so we must avoid
+                // sub-second precision for simple equality comparison
+                LogMetaPayload::Ping::clock::time_point{
+                    std::chrono::nanoseconds(123000000000)})),
+        makeNormalLogEntry(1, 4, "entry with somewhat larger payload"),
+        makeNormalLogEntry(2, 5, "foobar"),
+    });
+
+    auto res =
+        persistor->insert(log.getLogIterator(), LogPersistor::WriteOptions{})
+            .get();
+    ASSERT_TRUE(res.ok());
+    EXPECT_EQ(res.get(), 5);
+  }
+
+  void checkFileHeader(StreamReader& reader) {
+    auto header = reader.read<FileHeader>();
+    EXPECT_EQ(wMagicFileType, header.magic);
+    EXPECT_EQ(wCurrentVersion, header.version);
+  }
+
+  void checkLogEntry(StreamReader& reader, LogIndex idx, LogTerm term,
+                     RecordType type,
+                     std::variant<LogPayload, LogMetaPayload> payload) {
+    auto* data = reader.data();
+    auto dataSize = reader.size();
+
+    VPackBuilder builder;
+    VPackSlice payloadSlice;
+    if (std::holds_alternative<LogPayload>(payload)) {
+      auto const& p = get<LogPayload>(payload);
+      payloadSlice = p.slice();
+    } else {
+      auto const& p = get<LogMetaPayload>(payload);
+      p.toVelocyPack(builder);
+      payloadSlice = builder.slice();
+    }
+
+    auto expectedSize =
+        sizeof(Record::CompressedHeader) +
+        Record::paddedPayloadSize(payloadSlice.byteSize())  // payload
```

Comment:
I introduced a named constant for the alignment and added some test for `Record::paddedPayloadSize`. Would you still prefer to use a different calculation here?

Reply:
I'd prefer it because I think it then tests `paddedPayloadSize` as well. However, it's really an absolutely minor opinion; if you don't think it's worth it, just resolve the comment.